evennia/utils/evmenu.py

```diff
@@ -857,17 +933,75 @@ def goto(self, nodename, raw_string, **kwargs):
                                 exec_kwargs,
                             )
 
-        self.nodetext = self._format_node(nodetext, display_options)
+        if isinstance(nodedata, dict):
+            # Use the new method to generate nodetext
+
+            if 'options' in nodedata:
+                nodedata['options']['display'] = display_options
+                optiondict = nodedata['options']
+                if 'formatter' in optiondict:
+                    formatter = getattr(self, 'options_format_'
+                                        + optiondict['formatter'])
+                    if not callable(formatter):
+                        formatter = getattr(self, 'options_format_no_bars')
+                else:
+                    formatter = getattr(self, 'options_format_no_bars')
+                nodedata = self._safe_call(formatter, '', **nodedata)
```

Comment:
This default value should be generalized and not done this deep inside the code. The possible formats for the `formatter` field should IMO be one of the following: 

   - `formatter = "formattername"`  -> Translated to a call `EvMenu.*_formattername(**default_kwargs)`, the `default_kwargs` are the no-bars, etc we want to have as ready, easy-to-use options to the default formatter. 
   - `formatter = ("formattername", {kwargs}")` -> Translated to f `EvMenu.*_formattername(**kwargs)` for supplying custom kwargs, to the default formatter or to your own. 
   - `formatter = myformatter` -> This will called as `myformatter(caller, evmenuinstance)` - it doesn't make sense to have default kwargs for a custom formatter. 
   - `formatter = (myformatter, {kwargs})` -> Will be called as `myformatter(caller, evmenuinstance, **kwargs)`.

The kwargs of this could thus be checked easily at the top, with default kwargs set in one place and then used later in the code. It'd simplify this code a lot, with many fewer if statements.

Reply:
I can move the default formats up to the top easily enough.

I think I need a little more explanation on ``EvMenu.*_formattername(**default_kwargs)``. Are you proposing that ``*`` be something like 'text_format' and 'formattername' be a variable? I didn't think that would work in python and even if it did I thought the preferred form was ```formatter = getattr(self, <string>)```
```nodedata = formatter(**kwargs)```

I used your ``_safe_call`` function because I thought it would be preferable, though that does require me to pass something to fill in the ``callback`` value.

If the intent isn't to call a variable name I would refer you to the discussion below. I think using individual functions called by variable name would be easier on developers.

Reply:
With `EvMenu.*_formattername(**default_kwargs)` I just refer to a method string-name built from the string you pass to the `formatter=` key (two of the four possible forms above), and then passed through the `getattr(self, ...)`.

Reply:
So basically it is calling by a variable name (as discussed below) you just don't want to use ``_safe_call``. No problem.